web/lumpTypeView.ts

```diff
@@ -478,97 +484,238 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
+            // For each colormap
+            for (let mapIndex = 0; mapIndex < colormap.getMapCount(); mapIndex++) {
+                // For each color in the palette
+                for (let colorIndex = 0; colorIndex < playpal.colorsPerPalette; colorIndex++) {
+                    colorHelper.mapIndex = mapIndex;
+                    // Calculate position to draw
+                    const row = mapIndex * scaleY;
+                    const column = colorIndex * scaleX;
+                    // Get RGB string
+                    const rgb = colorHelper.getColorBGRA(colorIndex);
+                    const rgbHex = rgb.toString(16).substring(2).padEnd(6, "0");
+                    // Set style and fill color
+                    context.fillStyle = `#${rgbHex}`;
+                    context.fillRect(column, row, scaleX, scaleY);
+                }
+            }
+        }
+    });
+}
 
+export function LumpTypeViewColormapByMap(scaleX:number = 4, scaleY:number = 4) {
+    return new LumpTypeView({
+        name: "Colormap (by map)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            // 2 containers are needed in order to center the inner container
+            const outerContainer = util.createElement({
+                class: "lump-view-playpal",
+                appendTo: root
+            });
+            const innerContainer = util.createElement({
+                class: "lump-view-playpal-inner",
+                appendTo: outerContainer
+            });
+            const files = getWadFileList(lump);
+            // Playpal and Colormap
+            const playpal = files.getPlaypal();
+            const colormap = lumps.WADColorMap.from(lump);
+            const numColormaps = colormap.getMapCount();
+            // Used to decide which colormap to render to the canvas
+            let currentColormap = 0;
+            // Arrow buttons to navigate between colormaps
+            const nextArrow = util.createElement({
+                content: "»",
+                class: "lump-view-playpal-navbtn",
+                onleftclick: () => {
+                    // Go to next colormap. If at the end, go to first.
+                    currentColormap += 1;
+                    if (currentColormap >= numColormaps) {
+                        currentColormap = 0;
+                    }
+                    drawPalette(currentColormap);
+                }
+            });
+            const prevArrow = util.createElement({
+                content: "«",
+                class: "lump-view-playpal-navbtn",
+                onleftclick: () => {
+                    // Go to previous colormap. If at the first, go to last.
+                    currentColormap -= 1;
+                    if (currentColormap < 0) {
+                        currentColormap = numColormaps - 1;
+                    }
+                    drawPalette(currentColormap);
+                }
+            });
```

Comment:
Good enough for now but I will likely follow up with a commit that reduces code repetition between this and the very similar playpal DOM code by using common helper functions